peps/pep-0750.rst

```diff
@@ -382,6 +419,28 @@ Two instances of ``Interpolation`` are defined to be equal if their ``value``,
             )
 
 
+Template and Interpolation Hashing
+----------------------------------
+
+The ``Template`` and ``Interpolation`` types implement the ``__hash__()`` method
+roughly as follows:
+
+.. code-block:: python
+
+    class Template:
+        def __hash__(self) -> int:
+            return hash(self.args)
+
+    class Interpolation:
+        def __hash__(self) -> int:
+            return hash((self.value, self.expr, self.conv, self.format_spec))
+
+An ``Interpolation`` instance is hashable if its ``value`` attribute
+is also hashable. Likewise, a ``Template`` instance is hashable if
+all of its ``Interpolation`` instances are hashable. In all other cases,
+``TypeError`` is raised.
```

Comment:
I _kinda_ see why you're doing it this way, but it means that a hypothetical caching HTML template processor will have a hard time caching templates reliably (e.g. if a dict of attributes is given as in the example, the template can't be cached, at least its natural hash and eq won't let it). And in such use cases presumably the values in the cache are ignored -- when a new template is processed, *its* values are combined with the cached processed template.

Reply:
My expectation is that the `Template.strings` tuple will be the common cache key in these scenarios.

Reply:
So then I would consider dropping hash and eq.